modules/caddyhttp/caddyauth/hashes.go

```diff
@@ -15,17 +15,25 @@
 package caddyauth
 
 import (
-	"golang.org/x/crypto/bcrypt"
+	"errors"
 
 	"github.com/caddyserver/caddy/v2"
+	"golang.org/x/crypto/bcrypt"
 )
 
 func init() {
 	caddy.RegisterModule(BcryptHash{})
 }
 
+// defaultBcryptCost cost 14 strikes a solid balance between security, usability, and hardware performance
+const defaultBcryptCost = 14
+
 // BcryptHash implements the bcrypt hash.
-type BcryptHash struct{}
+type BcryptHash struct {
+	// cost is the bcrypt hashing difficulty factor (work factor).
+	// Higher values increase computation time and security.
+	cost int
```

Comment:
Configurable parts of module should be exported. If there's assumption of default value, we usually implement those checks in `Provision` method, which I believe at this point it's needed.

Reply:
Is the Provision method called in the CLI context? I pushed a struct with an exported Provision method, but it doesn't seem to be invoked.

Reply:
Oh, sorry 🤦🏼 I got confused with things. You're right, `Provision` is not called from the CLI to hash the password. It's also not needed in the comparison. You can revert that last commit 😅